### DISCUSSION TITLE
Use V3 $parent on function

### DISCUSSION THREAD

**Discussion by nam-co** [livewire/livewire#5822] [1. Help] [answered]
Hi, hope somebody can help me, 
1. is there a way to use $parent inside the php functions, I've tried $this->parent but is not working
2. is it possible for $parent to update a parent variable? maybe something like $parent.set('array.key.item.name', value);

Appreciate any help

**Comment by joshhanley** [comment]
@nam-co no, I don't think using `$parent` on the php side will work, only frontend.

But yeah I think maybe you should be able to update a parent variable, but you will need to test it to confirm.

Hope this helps!

**Reply by nam-co** [reply]
@joshhanley thanks for the answer, it would be great is it was possible, in your opinion what is the best way to update parent in V3? thank you all for all the work on V3

**Reply by joshhanley** [reply]
@nam-co I don't think that would be possible as all that interaction is on the front end. I would dispatch an event to the parent from the child to update it, if it's happening as part of a bigger method call on the child.

**Comment by danie-ramdhani** [comment]
you mean something like this?

```html
<!-- on child component -->
<button type="button" wire:click="$parent.setFooProperty">set foo property</button>
```

```php
// on parent component
public function setFooProperty()
{
    $this->foo = 'foo';
}
```

**Reply by nam-co** [reply]
Hi @danie-ramdhani thank's for the message, I meant update from the php functions

```
<!-- on child component -->

    public function something()
    {
        //Call parent->setFooProperties() from the code instead of the frontend
    }

```

**Reply by danie-ramdhani** [reply]
maybe in this documentation:
https://livewire.laravel.com/docs/events#listening-for-events-from-children-only

```php
use Livewire\Attributes\On;
use Livewire\Component;
 
class Dashboard extends Component
{
    #[On('childComponentDoSomething', fromChildren: true)] 
    public function setFooProperty()
    {
        // ...
    }
}
```

but i got error: `Unknown named parameter $fromChildren` XD

**Reply by joshhanley** [reply]
@danie-ramdhani yeah it seems like `$fromChildren` hasn't been implemented. Let's continue looking at that problem in #5886

**Reply by nam-co** [reply]
Hi @joshhanley thanks, yes I understand, @danie-ramdhani finding seems a lot cooler than the old emit/listen hope the livewire team finds a way, anyway thanks for the work, $this->js() is amazing 👍

**Reply by nam-co** [reply, marked as answer]
@danie-ramdhani #[On is working now 👍 , fromChildren: true is not needed
